src/instrumentation/sdk.go

```diff
@@ -19,496 +19,98 @@ package instrumentation
-	if idx == -1 {
-		optional := true
-		container.Env = append(container.Env, corev1.EnvVar{
-			Name: constants.EnvNewRelicLicenseKey,
-			ValueFrom: &corev1.EnvVarSource{
-				SecretKeyRef: &corev1.SecretKeySelector{
-					LocalObjectReference: corev1.LocalObjectReference{Name: "newrelic-key-secret"},
-					Key:                  "new_relic_license_key",
-					Optional:             &optional,
-				},
-			},
-		})
-	}
-	idx = getIndexOfEnv(container.Env, constants.EnvNewRelicLabels)
-	if idx == -1 {
-		container.Env = append(container.Env, corev1.EnvVar{
-			Name:  constants.EnvNewRelicLabels,
-			Value: "operator:auto-injection",
-		})
-	}
-	idx = getIndexOfEnv(container.Env, constants.EnvNewRelicK8sOperatorEnabled)
-	if idx == -1 {
-		container.Env = append(container.Env, corev1.EnvVar{
-			Name:  constants.EnvNewRelicK8sOperatorEnabled,
-			Value: "true",
-		})
-	}
-	return pod
+// SdkInjector is used to inject our instrumentation into a pod
+type SdkInjector interface {
+	Inject(ctx context.Context, insts []*v1alpha2.Instrumentation, ns corev1.Namespace, pod corev1.Pod) corev1.Pod
 }
 
-func chooseServiceName(pod corev1.Pod, resources map[string]string, index int) string {
-	if name := resources[string(semconv.K8SDeploymentNameKey)]; name != "" {
-		return name
-	}
-	if name := resources[string(semconv.K8SStatefulSetNameKey)]; name != "" {
-		return name
-	}
-	if name := resources[string(semconv.K8SJobNameKey)]; name != "" {
-		return name
-	}
-	if name := resources[string(semconv.K8SCronJobNameKey)]; name != "" {
-		return name
-	}
-	if name := resources[string(semconv.K8SPodNameKey)]; name != "" {
-		return name
-	}
-	return pod.Spec.Containers[index].Name
+// NewrelicSdkInjector is the base struct used to inject our instrumentation into a pod
+type NewrelicSdkInjector struct {
+	client           client.Client
+	logger           logr.Logger
+	injectorRegistry *apm.InjectorRegistery
 }
 
-// obtains version by splitting image string on ":" and extracting final element from resulting array.
-func chooseServiceVersion(pod corev1.Pod, index int) string {
-	parts := strings.Split(pod.Spec.Containers[index].Image, ":")
-	tag := parts[len(parts)-1]
-	//guard statement to handle case where image name has a port number
-	if strings.Contains(tag, "/") {
-		return ""
+// NewNewrelicSdkInjector is used to create our injector
+func NewNewrelicSdkInjector(logger logr.Logger, client client.Client, injectorRegistry *apm.InjectorRegistery) *NewrelicSdkInjector {
+	return &NewrelicSdkInjector{
+		client:           client,
+		logger:           logger,
+		injectorRegistry: injectorRegistry,
 	}
-	return tag
 }
 
-func resourceMapToStr(res map[string]string) string {
-	keys := make([]string, 0, len(res))
-	for k := range res {
-		keys = append(keys, k)
-	}
-	sort.Strings(keys)
-
-	var str = ""
-	for _, k := range keys {
-		if str != "" {
-			str += ","
-		}
-		str += fmt.Sprintf("%s=%s", k, res[k])
+// Inject is used to inject our instrumentation's into a pod
```

Comment:
inject our instrumentaion CRs into a pod?

Reply:
updated the description